elobot.py

```diff
@@ -11,195 +11,254 @@
+        Return (winner elo delta, loser elo delta)
+        """
+        if not match.pending:
+            raise ValuError("Match must be pending to apply.")
+
+        with db.transaction():
+            winner = self.players[match.winner_handle]
+            loser  = self.players[match.loser_handle]
+            winner_elo_delta, loser_elo_delta = self.rank_game(winner, loser)
+
+            match.pending = False
+            match.save()
+
+        return (winner_elo_delta, loser_elo_delta)
+
     def init_players(self):
-        print('initializing in-memory player objects')
-        print('loading match history')
+        """Initializes self.players with the games stored in the database"""
+        print('Initializing players...')
         matches = list(Match.select().order_by(Match.id))
-        print('simulating games...')
         for match in matches:
+            print('Recapping match: {}'.format(match.__dict__))
             if not match.pending:
-                winner, loser = self.players[match.winner_handle], self.players[match.loser_handle]
+                winner = self.players[match.winner_handle]
+                loser  = self.players[match.loser_handle]
                 self.rank_game(winner, loser)
-        print('finished initializing players')
-        print('players: {}'.format([ str(player) for player in self.players.values() ]))
-
+        print('Player initialization complete.')
 
-    def connect(self):
-        print('connecting!')
+    def ensure_connected(self):
         sleeptime = 0.1
-
-        while True:
+        while not self.slack_client.server.connected:
+            print('Was disconnected, attemping to reconnect...')
             try:
                 self.slack_client.rtm_connect()
-                break
-            except:
-                time.sleep(sleeptime)
-
-                #Exponential back off with a max wait of 30s
-                if sleeptime < 30:
-                    sleeptime *= 2
+            except:  # TODO: Except what
+                pass
+            time.sleep(sleeptime)
+            sleeptime = min(30, sleeptime * 2)  # Exponential back off with a max wait of 30s
 
     def heartbeat(self):
+        """Send a heartbeat if necessary"""
         now = int(time.time())
         if now > self.last_ping + 3:
             self.slack_client.server.ping()
             self.last_ping = now
 
     def talk(self, message):
-        #print(message)
-        self.slack_client.api_call('chat.postMessage', channel=self.channel, text=message, username=self.config['bot_name'])
+        """Send a message to the Slack channel"""
+        self.slack_client.api_call('chat.postMessage', channel=self.channel_id, text=message, username=self.name)
 
-    def run(self):
-        print('running!')
-        #self.talk(self.config['bot_name'] + ' online!')
+    def talk_to(self, user_id, message):
+        message = message[0].lower() + message[1:]
```

Comment:
why do you do this? this is rather arcane...

Reply:
Because `talk_to(person, message)` conceptually embodies that we're saying some `message` to some `person`. That `message` is a fully-formed, complete, and independent idea. As an effect of this, it should have correct grammar, including a capitalized first letter of the sentence.

In `talk_to`, we combine the two arguments by smushing them together into one string a la `Person: message`. Since the message is now a part of a larger sentence, we should lowercase the first letter.